### PR TITLE
Create client to interface with Codespaces

### PR DESCRIPTION
**Pull request by ericdyoon**
Requires https://github.com/dotnet/project-system/pull/6563 to be merged in first

Frontend will now be the only project in project-system-tools.
Extension will be installed alongside project-system version in https://github.com/dotnet/project-system/pull/6563

Attached Issues:
Shared project bug: Fixes https://github.com/dotnet/project-system-tools/issues/366

Creates Issues:
Duplicated Shared project: https://github.com/dotnet/project-system-tools/issues/370

### CODE REVIEW

File: ProjectSystemTools.sln [hunk 1]
```diff
@@ -1,13 +1,11 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.26913.0
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.30421.23
```
Comment (dmonroym) [inline ProjectSystemTools.sln:4]: Where is this version number coming from?

File: src/LogModel/Microsoft.VisualStudio.ProjectSystemTools.Shared.csproj [hunk 2]
```diff
@@ -27,4 +30,4 @@
     </EmbeddedResource>
   </ItemGroup>
```
Comment (dmonroym) [inline src/LogModel/Microsoft.VisualStudio.ProjectSystemTools.Shared.csproj:31]: Remove?

**Comment by dmonroym** [comment]
> <?xml version="1.0" encoding="utf-8"?>

Delete these xlf

---
Refers to: src/LogModel/xlf/BuildLoggingResources.cs.xlf:1 in 76f54f0. [](commit_id = 76f54f07084adf160563053b94af9610000b2fd6, deletion_comment = False)

**Comment by dmonroym** [comment]
Rename LogModel folder to Microsoft.VisualStudio.ProjectSystemTools.Shared

File: src/ProjectSystemTools/BrokeredService.pkgdef [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[$RootKey$\BrokeredServices\LoggerService\1.0]
+@="{6BEB9F10-4E64-41B7-B4E3-1BE77CD63DD4}"
+"audience"=dword:00000503
```
Comment (dmonroym) [inline src/ProjectSystemTools/BrokeredService.pkgdef:3]: What is this pkgdef doing?

File: src/ProjectSystemTools/Microsoft.VisualStudio.ProjectSystemTools.FrontEnd.csproj [hunk 4]
```diff
@@ -120,10 +123,10 @@
     <Compile Update="BinaryLogEditor\LogViewControl.xaml.cs">
       <DependentUpon>LogViewControl.xaml</DependentUpon>
     </Compile>
-    <Compile Update="RoslynLogging\RoslynLoggingResources.Designer.cs">
+    <Compile Update="LogModel\Resources.Designer.cs">
+      <DependentUpon>Resources.resx</DependentUpon>
```
Comment (dmonroym) [inline src/ProjectSystemTools/Microsoft.VisualStudio.ProjectSystemTools.FrontEnd.csproj:127]: Don't think this is necessary

File: src/ProjectSystemTools/Microsoft.VisualStudio.ProjectSystemTools.FrontEnd.csproj [hunk 5]
```diff
@@ -105,6 +105,9 @@
       <Generator>MSBuild:Compile</Generator>
     </Page>
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\LogModel\Microsoft.VisualStudio.ProjectSystemTools.Shared.csproj" />
```
Comment (dmonroym) [inline src/ProjectSystemTools/Microsoft.VisualStudio.ProjectSystemTools.FrontEnd.csproj:109]: You'll have to change this reference if you change the folder name

File: src/ProjectSystemTools/BinaryLogEditor/ViewModel/BuildViewModel.cs [hunk 6]
```diff
@@ -8,7 +8,7 @@ namespace Microsoft.VisualStudio.ProjectSystem.Tools.BinaryLogEditor.ViewModel
 {
     internal sealed class BuildViewModel : NodeViewModel
     {
-        private readonly LogModel.Build _build;
+        private readonly ProjectSystem.LogModel.Build _build;
```
Comment (dmonroym) [inline src/ProjectSystemTools/BinaryLogEditor/ViewModel/BuildViewModel.cs:11]: Related to https://github.com/dotnet/project-system/pull/6563#discussion_r482493732
I would consider renaming this data type

File: src/ProjectSystemTools/LogModel/Builder/EvaluationInfo.cs [hunk 7]
```diff
@@ -3,6 +3,7 @@
 using System.Collections.Generic;
 using System.Linq;
 using Microsoft.Build.Framework;
+using Microsoft.VisualStudio.ProjectSystem.Tools.LogModel;
```
Comment (dmonroym) [inline src/ProjectSystemTools/LogModel/Builder/EvaluationInfo.cs:6]: Why was this added?

File: src/ProjectSystemTools/LogModel/Builder/ItemInfo.cs [hunk 8]
```diff
@@ -3,6 +3,7 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using Microsoft.Build.Framework;
+using Microsoft.VisualStudio.ProjectSystem.Tools.LogModel;
 
 namespace Microsoft.VisualStudio.ProjectSystem.LogModel.Builder
```
Comment (dmonroym) [inline src/ProjectSystemTools/LogModel/Builder/ItemInfo.cs:8]: Remove?

File: src/ProjectSystemTools/LogModel/Builder/ModelBuilder.cs [hunk 9]
```diff
@@ -10,6 +10,7 @@
 using System.Text.RegularExpressions;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Framework.Profiler;
+using Microsoft.VisualStudio.ProjectSystem.Tools.LogModel;
```
Comment (dmonroym) [inline src/ProjectSystemTools/LogModel/Builder/ModelBuilder.cs:13]: Remove?

File: src/ProjectSystemTools/LogModel/Builder/ProjectInfo.cs [hunk 10]
```diff
@@ -4,6 +4,7 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using Microsoft.Build.Framework;
+using Microsoft.VisualStudio.ProjectSystem.Tools.LogModel;
```
Comment (dmonroym) [inline src/ProjectSystemTools/LogModel/Builder/ProjectInfo.cs:7]: Remove

File: src/ProjectSystemTools/LogModel/Builder/TargetInfo.cs [hunk 11]
```diff
@@ -4,6 +4,7 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using Microsoft.Build.Framework;
+using Microsoft.VisualStudio.ProjectSystem.Tools.LogModel;
```
Comment (dmonroym) [inline src/ProjectSystemTools/LogModel/Builder/TargetInfo.cs:7]: Remove

File: src/ProjectSystemTools/LogModel/Builder/TaskInfo.cs [hunk 12]
```diff
@@ -4,6 +4,7 @@
 using System.Collections.Generic;
 using System.Linq;
 using Microsoft.Build.Framework;
+using Microsoft.VisualStudio.ProjectSystem.Tools.LogModel;
```
Comment (dmonroym) [inline src/ProjectSystemTools/LogModel/Builder/TaskInfo.cs:7]: Remove

**Review by dmonroym — CHANGES_REQUESTED** [review]
:clock1:

File: src/ProjectSystemTools/BrokeredService.pkgdef (hunk 3 above)
Comment (ericdyoon) [reply src/ProjectSystemTools/BrokeredService.pkgdef:3]: This is needed for the frontend to ask for the Codespaces service from project-system

File: src/ProjectSystemTools/LogModel/Builder/EvaluationInfo.cs (hunk 7 above)
Comment (ericdyoon) [reply src/ProjectSystemTools/LogModel/Builder/EvaluationInfo.cs:6]: It looks like it was because I didn't change namespaces of files while I was rearranging the projects, so it couldn't find the Resources.resx file without this using statement.

File: ProjectSystemTools.sln (hunk 1 above)
Comment (ericdyoon) [reply ProjectSystemTools.sln:4]: Should be changed back now, didn't seem to affect the repo

File: src/LogModel/Microsoft.VisualStudio.ProjectSystemTools.Shared.csproj (hunk 2 above)
Comment (ericdyoon) [reply src/LogModel/Microsoft.VisualStudio.ProjectSystemTools.Shared.csproj]: Done :)

File: src/ProjectSystemTools/BinaryLogEditor/ViewModel/BuildViewModel.cs (hunk 6 above)
Comment (ericdyoon) [reply src/ProjectSystemTools/BinaryLogEditor/ViewModel/BuildViewModel.cs]: Should be renamed now :)

File: src/ProjectSystemTools/LogModel/Builder/EvaluationInfo.cs (hunk 7 above)
Comment (ericdyoon) [reply src/ProjectSystemTools/LogModel/Builder/EvaluationInfo.cs]: Renamed the namespace, so I was able to remove these using statements

**Review by jjmew — APPROVED** [review]
:shipit:

**Comment by MiYanni** [comment]
Closing this for the same reason as mentioned in https://github.com/dotnet/project-system/pull/6563. We can reopen and investigate this transition when this endeavor becomes a priority.
